plugins/cros-ec/fu-cros-ec-firmware.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-fmap-firmware.h"
+#include "fu-cros-ec-common.h"
+#include "fu-cros-ec-firmware.h"
+
+#define MAXSECTIONS	2
+
+struct _FuCrosEcFirmware {
+	FuFirmware			parent_instance;
+	struct cros_ec_version		version;
+	FuCrosEcFirmwareSection		sections[MAXSECTIONS];
+};
+
+G_DEFINE_TYPE (FuCrosEcFirmware, fu_cros_ec_firmware, FU_TYPE_FIRMWARE)
+
+static gboolean
+fu_cros_ec_firmware_parse (FuFirmware *firmware,
+			   GBytes *fw,
+			   guint64 addr_start,
+			   guint64 addr_end,
+			   FwupdInstallFlags flags,
+			   GError **error)
+{
+	FuCrosEcFirmware *self = FU_CROS_EC_FIRMWARE (firmware);
+	g_autoptr(FuFirmware) fmap_firmware = fu_fmap_firmware_new ();
+
+	self->sections[0].name = "RO";
+	self->sections[1].name = "RW";
+
+	if (!fu_firmware_parse (fmap_firmware, fw, flags, error))
+		return FALSE;
```

Comment:
this specifically should go away when you do the subclass call

Reply:
Agree.

Reply:
Done.